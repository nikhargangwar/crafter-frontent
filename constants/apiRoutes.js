const userRoute = 'api/user/';

export default { userRoute };