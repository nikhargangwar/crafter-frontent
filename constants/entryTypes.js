const entryTypes = {
  userConfig :{
    FRONTEND:{
      port:'integer',
      numberOfReplicas : 'integer',
      name:'string',
    },
    BACKEND:{
      port:'integer',
      numberOfReplicas:'integer',
      name:'string',
    },
    DATABASE:{
      port:'integer',
      numberOfReplicas:'integer',
      dbUser:'string',
      dbPassword:'string',
      name:'string',
      schemaName :'string',
    }},
  dockerConfig:{
    username:'string',
    email:'e-mail',
    repositoryImageAddress : 'string',
    token:'string'},
};

export default entryTypes;
