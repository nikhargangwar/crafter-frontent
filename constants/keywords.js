const arrayOfWordsForLandingPage = [ ' containerized applications on Kubernetes with our intuitive K8s Designer tool.', ' Frontend - React.', ' Backend - Node.js.', ' Database - Postgres.' ];

export default arrayOfWordsForLandingPage;