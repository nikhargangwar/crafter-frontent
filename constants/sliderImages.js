import loignSliderImg1 from '../assets/undraw_code_thinking_re_gka2.svg';
import loginSliderImg2 from '../assets/undraw_dev_focus_re_6iwt.svg';
import loginSliderImg3 from '../assets/undraw_specs_re_546x.svg';
import loginSliderImg4 from '../assets/undraw_coding_re_iv62.svg';
import signupSliderImg1 from '../assets/undraw_programmer_re_owql.svg';
import signupSliderImg2 from '../assets/undraw_programming_re_kg9v.svg';
import signupSliderImg3 from '../assets/undraw_dev_productivity_re_fylf.svg';
import signupSliderImg4 from '../assets/undraw_in_the_office_re_jtgc.svg';


const loginImages = [
  {
    original: loignSliderImg1
  },
  {
    original: loginSliderImg2
  },
  {
    original: loginSliderImg3
  },
  {
    original: loginSliderImg4
  }
];

const signupImages = [
  {
    original: signupSliderImg1
  },
  {
    original: signupSliderImg2
  },
  {
    original: signupSliderImg3
  },
  {
    original: signupSliderImg4
  }
];

export default { loginImages, signupImages };