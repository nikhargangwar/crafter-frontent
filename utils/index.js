import CustomNode from './CustomNode';
import FormValidation from './FormValidation';
import makeRequest from './makeRequest';
import {camelize, getConfiguration, getSpaceSeparated, updateConfiguration} from './ServiceHandler';
export { CustomNode, FormValidation, makeRequest, camelize, getConfiguration, getSpaceSeparated, updateConfiguration };