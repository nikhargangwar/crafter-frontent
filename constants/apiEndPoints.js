import apiRoutes from './apiRoutes';

const GET_ALL_BLUEPRINTS = () => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: '/blueprints',
  method: 'GET',
});

const GET_BLUEPRINT_METADATA = (blueprintId) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/blueprints/${blueprintId}`,
  method: 'GET',
});

const GET_IMAGES =(name)=>({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/minio/images/${name}`,
  method: 'GET',
});

const GET_SERVICES_BY_BLUEPRINT_ID = (blueprintId, version) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/blueprints/${blueprintId}/versions/${version}/services`,
  method: 'GET',
});

const CREATE_BLUEPRINT = () => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: '/blueprints',
  method: 'POST',
});

const DELETE_BLUEPRINT = (blueprintId) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/blueprints/${blueprintId}`,
  method: 'DELETE',
});

const CREATE_NEW_SERVICE =(blueprintId, version) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/blueprints/${blueprintId}/versions/${version}/services`,
  method: 'POST'
});

const UPDATE_SERVICE_POSITION= (blueprintId, serviceId, version) =>({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/blueprints/${blueprintId}/versions/${version}/services/${serviceId}`,
  method: 'PUT',
});

const registerUser = (name, otp, email, password) => {
  return {
    baseURL: process.env.REACT_APP_API_GATEWAY_URL,
    url: apiRoutes.userRoute + 'register',
    method: 'post',
    data: {
      name,
      otp,
      email,
      password,
    }
  };
};

const loginUser = (email, password) => {
  return {
    baseURL: process.env.REACT_APP_API_GATEWAY_URL,
    url: apiRoutes.userRoute + 'login',
    method: 'post',
    data: {
      email,
      password,
    }
  };
};

const validateUser = (token) => {
  return {
    baseURL: process.env.REACT_APP_API_GATEWAY_URL,
    url: apiRoutes.userRoute + 'validate',
    method: 'get',
    headers: {
      authorization: token
    }
  };

};

const generateOtp = (email) => {
  return {
    baseURL: process.env.REACT_APP_API_GATEWAY_URL,
    url: apiRoutes.userRoute + 'generate/otp',
    method: 'post',
    data: {
      email,
    }
  };
};

const GET_SYSTEM_COMPONENTS = () => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: '/plugins/types/SystemComponent',
  method: 'GET',
});

const GET_DEPLOYMENT_STACK = () => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: '/plugins/types/DeploymentStack',
  method: 'GET',
});

const GET_PLUGIN_BY_ID = (pluginId) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/plugin/${pluginId}`,
  method: 'GET',
});


const CONNECT_SERVICES = (blueprintId,serviceId,version) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/blueprints/${blueprintId}/versions/${version}/services/${serviceId}/connections`,
  method: 'POST',

});

const DISCONNECT_SERVICES = (blueprintId,serviceId,version) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/blueprints/${blueprintId}/versions/${version}/services/${serviceId}/connections`,
  method: 'DELETE',
});

const GET_SERVICE_BY_ID = (blueprintId, serviceId, version) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/blueprints/${blueprintId}/versions/${version}/services/${serviceId}`,
  method: 'GET',
});

const UPDATE_CONFIGURATIONS = (blueprintId, serviceId, version) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/blueprints/${blueprintId}/versions/${version}/services/${serviceId}/configurations`,
  method: 'PUT',
});

const UPDATE_METADATA = (blueprintId, serviceId, version) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/blueprints/${blueprintId}/versions/${version}/services/${serviceId}/metadata`,
  method: 'PUT',
});

const DELETE_SERVICE_BY_ID = (blueprintId, serviceId, version) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/blueprints/${blueprintId}/versions/${version}/services/${serviceId}`,
  method: 'DELETE',
});

const UPDATE_BLUEPRINT_METADATA = (blueprintId) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/blueprints/${blueprintId}`,
  method: 'PUT',
});

const SEARCH_CONFIGURATIONS = (blueprintId, serviceId, version) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/blueprints/${blueprintId}/versions/${version}/services/${serviceId}/configurations`,
  method: 'GET',
});

const GET_ALL_CLUSTERS = () => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: '/clusters',
  method: 'GET',
});

const GET_CLUSTER_BY_ID = (clusterId) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/clusters/${clusterId}`,
  method: 'GET',
});

const CREATE_CLUSTER = () => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: '/clusters',
  method: 'POST',
});

const GET_ALL_TASKS= () => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: '/tasks',
  method: 'GET',
  
});

const GET_TASKS_BY_ID = (id) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/tasks/${id}`,
  method: 'GET',
});

const DOWNLOAD_BOILERPLATE = (objectName) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/minio/boilerplate/${objectName}`,
  method: 'GET',
});

const DELETE_CLUSTER = (clusterId) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/clusters/${clusterId}`,
  method: 'DELETE',
});

const EXPORT_BLUEPRINT = (blueprintId, version) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/blueprints/${blueprintId}/versions/${version}/exports`,
  method: 'POST',
});

const DEPLOY_CLUSTER = (clusterId,blueprintId, version) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/clusters/${clusterId}/blueprints/${blueprintId}/versions/${version}/deploy`,
  method: 'POST',
});

const TEST_CONNECTION = (clusterId) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/clusters/${clusterId}/testConnection`,
  method: 'POST',
});

export default { 
  generateOtp,
  registerUser,
  loginUser,
  validateUser,
  GET_ALL_BLUEPRINTS,
  GET_BLUEPRINT_METADATA,
  CREATE_BLUEPRINT,
  DELETE_BLUEPRINT,
  GET_SYSTEM_COMPONENTS,
  GET_SERVICES_BY_BLUEPRINT_ID,
  GET_SERVICE_BY_ID,
  UPDATE_CONFIGURATIONS,
  UPDATE_METADATA,
  CREATE_NEW_SERVICE,
  DELETE_SERVICE_BY_ID,
  CONNECT_SERVICES,
  DISCONNECT_SERVICES,
  UPDATE_BLUEPRINT_METADATA,
  UPDATE_SERVICE_POSITION,
  SEARCH_CONFIGURATIONS,
  GET_ALL_CLUSTERS,
  GET_CLUSTER_BY_ID,
  CREATE_CLUSTER,
  GET_DEPLOYMENT_STACK,
  GET_ALL_TASKS,
  GET_TASKS_BY_ID,
  DOWNLOAD_BOILERPLATE,
  GET_IMAGES,
  GET_PLUGIN_BY_ID,
  DELETE_CLUSTER,
  EXPORT_BLUEPRINT,
  DEPLOY_CLUSTER,
  TEST_CONNECTION
};
