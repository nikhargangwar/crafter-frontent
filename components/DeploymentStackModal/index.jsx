import React, { useEffect, useState } from 'react';
import { Box, Button, Card, Checkbox, FormControl, FormHelperText, Grid, InputAdornment, InputLabel, Link, ListItemText, MenuItem, Modal, OutlinedInput, Select, TextField, Tooltip, Typography } from '@mui/material';
import { Storage, Close, Search, Info } from '@mui/icons-material';
import PropTypes from 'prop-types';
import { makeRequest } from '../../utils';
import { apiEndPoints } from '../../constants';
import { useNavigate } from 'react-router-dom';
import { modalStyle } from '../../constants/styles';

import { capitalizeEachWord, separatePascalCase } from '../../utils/ServiceHandler';
function DeploymentStackModal({openModal, setOpenModal, isEdit}) {
  const [components, setComponents] = useState([]);
  const [activeStep, setActiveStep] = useState(0);
  const [blueprintName, setBlueprintName] = useState('');
  const [blueprintDescription, setBlueprintDescription] = useState('');
  const [deploymentStackPlugins, setDeploymentStackPlugins] = useState([]);
  const [selectedDeploymentStack, setSelectedDeploymentStack] = useState([]);
  const [query, setQuery] = useState('');
  const navigate = useNavigate();
  const token = localStorage.getItem('jwt_token');

  useEffect(() => {
    makeRequest.makeRequest(apiEndPoints.GET_SYSTEM_COMPONENTS(), navigate, token).then((response) => {
      setComponents(response);
    });
    makeRequest.makeRequest(apiEndPoints.GET_DEPLOYMENT_STACK(), navigate, token). then((response) => {
      setDeploymentStackPlugins(response);
      setSelectedDeploymentStack(response.filter((item) => item.metadata.name === 'DockerCompose' || item.metadata.name === 'KubernetesManifest'));
    });
  }, []);
  
  const dockerPlugin = deploymentStackPlugins.find((item) => item.metadata.name === 'DockerCompose');
  const kubernetesPlugin = deploymentStackPlugins.find((item) => item.metadata.name === 'KubernetesManifest');

  const handleDeploymentStackChange = (event) => {
    const { target: { value } } = event;
    setSelectedDeploymentStack(
      typeof value === 'string' ? value.split(',') : value,
    );
  };

  return (
    <Modal
      open={openModal}
    >
      <Box sx={modalStyle}>
        <div className='flex justify-between items-center'>
          <Typography variant='h5' style={{ fontWeight:'bold' }}>{isEdit && 'Edit '}{(activeStep || isEdit) ? 'Blueprint Information' : 'Deployment Stack' }</Typography>
          <Close onClick={() => {
            setOpenModal(false);
            setActiveStep(0);
          }}
          className='cursor-pointer' />
        </div>
        {activeStep === 0 && !isEdit && <>
          <FormControl required variant='outlined' fullWidth sx={{my:2}}>
            <InputLabel sx={{color: 'white'}} id='select-deployment-stack'>Select Deployment Stack</InputLabel>
            <Select
              labelId='select-deployment-stack'
              multiple
              fullWidth
              input={<OutlinedInput label="Select Deployment Stack" />}
              value={selectedDeploymentStack.length <= 2 ? [dockerPlugin, kubernetesPlugin] : selectedDeploymentStack}
              onChange={handleDeploymentStackChange}
              renderValue={(selected) => selected.map(stack => separatePascalCase(stack?.metadata.name)).join(', ')}
              MenuProps={{
                style: {
                  maxHeight: 300,
                  width: '100%',
                }
              }}
              sx={{color: 'white', fieldset : {borderColor: 'white'}}}
            >
              {deploymentStackPlugins.map((item) => (
                <MenuItem key={item.id} value={item} >
                  <ListItemText primary={separatePascalCase(item.metadata.name)} />
                  <Checkbox checked={selectedDeploymentStack.indexOf(item) > -1 || item === dockerPlugin || item === kubernetesPlugin}/>
                </MenuItem>
              ))}
            </Select>
            <FormHelperText sx={{color: 'white'}}>Docker Compose and Kubernetes Manifest Files will always be generated</FormHelperText>
          </FormControl>
          <Box className='flex justify-between items-center gap-x-10'>
            <Typography variant='h6' style={{ fontWeight: 'bold' }}>Available Components for Dapr </Typography>
            <TextField placeholder='Search' variant="outlined" size='small' type='search'
              onChange={(e) => setQuery(e.target.value)}
              InputProps={{
                startAdornment: (
                  <InputAdornment position="start">
                    <Search />
                  </InputAdornment>
                )
              }} />
          </Box>
          <Link href='#'>
            <Typography>View all components available for Dapr</Typography>
          </Link>
          <Typography style={{ fontWeight: 'bold', marginTop: 15 }}>Common Components</Typography>
          <Grid container spacing={3}>
            {components
              .filter((component) => {
                if (query === '') {
                  return true;
                }
                return component.metadata.name.toLowerCase().includes(query.toLowerCase());
              } )
              .map((component) => (
                <Grid item xs={12} sm={6} md={4} key={component.id}>
                  <Card className='my-4 relative'>
                    <Tooltip title={component.metadata.description}>
                      <Info color='primary' fontSize='small' className='absolute right-0 m-2' />
                    </Tooltip>
                    <div className='flex p-5 items-center gap-2'>
                      <Storage color='primary' fontSize='large' />
                      <div className='flex flex-col'>
                        <Typography fontWeight='bold'>{capitalizeEachWord(component.metadata.name.replace(/-/, ' '))}</Typography>
                      </div>
                    </div>
                  </Card>
                </Grid>
              ))}
          </Grid>
          <Button variant='contained' className='w-full' sx={{ bgcolor: '#18CCCC', color: 'white', fontWeight: 'bold', fontSize: '16px' }} onClick={() => setActiveStep((prevActiveStep) => prevActiveStep + 1 )}>Continue</Button>
        </>}
        {(activeStep===1 || isEdit) && <> 
          <Box 
            component='form'
            noValidate
            autoComplete='off'
          >
            <TextField
              label='Enter Blueprint Name'
              variant='outlined'
              fullWidth
              margin='normal'
              required
              onChange={(e) => setBlueprintName(e.target.value)}
            />
            <TextField
              label='Enter Blueprint Description'
              variant='outlined'
              fullWidth
              margin='normal'
              multiline
              rows={3}
              required
              onChange={(e) => setBlueprintDescription(e.target.value)}
            />
            <Button
              variant='contained'
              className='w-full'
              sx={{ bgcolor: '#18CCCC', color: 'white', fontWeight: 'bold', fontSize: '16px' }}
              disabled={!blueprintName || !blueprintDescription}
              onClick={() => {
                makeRequest.makeRequest(apiEndPoints.CREATE_BLUEPRINT(), navigate, token,
                  {
                    data: {
                      blueprintName,
                      deploymentStackPlugins: selectedDeploymentStack.length <= 2 ? [dockerPlugin, kubernetesPlugin].map(item => item?.id): selectedDeploymentStack.map((item) => item?.id),
                      metadata: {
                        name: blueprintName,
                        description: blueprintDescription,
                        createdAt: new Date().toLocaleDateString(),
                        lastUpdated: new Date().toLocaleDateString(),
                        author: JSON.parse(localStorage.getItem('user_detail')).fullName,
                      }
                    }
                  }
                ).then((response) => {
                  navigate('/dashboard/' + response.id + '/v/' + response.version);
                });
              }}
            >
              {isEdit ? 'Save': 'Create'}
            </Button> 
          </Box>
        </>}
      </Box>
    </Modal>
  );
}

export default DeploymentStackModal;

DeploymentStackModal.propTypes = {
  openModal: PropTypes.bool.isRequired,
  setOpenModal: PropTypes.func.isRequired,
  isEdit: PropTypes.bool
};
