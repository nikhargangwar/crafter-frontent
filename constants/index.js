import apiEndPoints from './apiEndPoints';
import apiRoutes from './apiRoutes';
import entryTypes from './entryTypes';
import arrayOfWordsForLandingPage from './keywords';
import sliderImages from './sliderImages';

export { apiEndPoints, apiRoutes, entryTypes, arrayOfWordsForLandingPage, sliderImages };
