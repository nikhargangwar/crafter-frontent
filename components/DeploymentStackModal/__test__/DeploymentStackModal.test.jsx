import React from 'react';
import { fireEvent, render, screen} from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import DeploymentStackModal from '..';
import { makeRequest } from '../../../utils';
import { apiEndPoints } from '../../../constants';

const mockedNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedNavigate,
}));

beforeEach(() => {
  jest.spyOn(makeRequest, 'makeRequest').mockResolvedValue([
    {
      id: 1,
      metadata: {
        name: 'test name 1',
        description: 'test description 1',
      }
    },
    {
      id: 2,
      metadata: {
        name: 'test name 2',
        description: 'test description 2',
      }
    }
  ]);
});

describe('DeploymentStackModal component', () => {
  const setOpenModal = jest.fn();
  const openModal = true;

  it('should render correctly', () => {
    const { asFragment } = render(<Router><DeploymentStackModal setOpenModal={setOpenModal} openModal={openModal} /></Router>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render all components', async () => {
    render(<Router><DeploymentStackModal setOpenModal={setOpenModal} openModal={openModal} /></Router>);
    const names = await screen.findAllByText(/test name/i);
    expect(names).toHaveLength(2);
  });

  it('should call setOpenModal when Close button is clicked', async () => {
    render(<Router><DeploymentStackModal setOpenModal={setOpenModal} openModal={openModal} /></Router>);
    const closeButton = screen.getByTestId('CloseIcon');
    await fireEvent.click(closeButton);
    expect(setOpenModal).toHaveBeenCalledWith(false);
  });

  it('should display next page when Continue button is clicked', async () => {
    render(<Router><DeploymentStackModal setOpenModal={setOpenModal} openModal={openModal} /></Router>);
    const continueButton = screen.getByRole('button', { name: 'Continue' });
    expect(continueButton).toBeTruthy();
    await fireEvent.click(continueButton);
    const nameTextField = screen.getByLabelText(/enter blueprint name/i);
    expect(nameTextField).toBeTruthy();
    const descriptionTextArea = screen.getByLabelText(/enter blueprint description/i);
    expect(descriptionTextArea).toBeTruthy(); 
  });

  it('should send a request to create a new blueprint when Create button is clicked', async () => {
    const mockedDate = new Date(2021, 1, 1);
    jest.spyOn(global, 'Date').mockImplementation(() => mockedDate);
    jest.spyOn(JSON, 'parse').mockReturnValue({ fullName: 'test name' });
    render(<Router><DeploymentStackModal setOpenModal={setOpenModal} openModal={openModal} /></Router>);
    const continueButton = screen.getByRole('button', { name: 'Continue' });
    await fireEvent.click(continueButton);
    const nameTextField = screen.getByLabelText(/enter blueprint name/i);
    await fireEvent.change(nameTextField, { target: { value: 'test name' } });
    const descriptionTextArea = screen.getByLabelText(/enter blueprint description/i);
    await fireEvent.change(descriptionTextArea, { target: { value: 'test description' } });
    const createButton = screen.getByRole('button', { name: 'Create' });
    await fireEvent.click(createButton);
    expect(makeRequest.makeRequest).toHaveBeenLastCalledWith(apiEndPoints.CREATE_BLUEPRINT(), mockedNavigate, null, {
      data: {
        blueprintName: 'test name',
        deploymentStackPlugins: [ undefined, undefined ],
        metadata: {
          name: 'test name',
          description: 'test description',
          createdAt: mockedDate.toLocaleDateString(),
          lastUpdated: mockedDate.toLocaleDateString(),
          author: 'test name'
        },
      }
    });
    expect(makeRequest.makeRequest).toHaveBeenCalledTimes(3);
  });
});
