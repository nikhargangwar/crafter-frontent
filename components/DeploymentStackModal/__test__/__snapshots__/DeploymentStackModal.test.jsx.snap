// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeploymentStackModal component should render correctly 1`] = `<DocumentFragment />`;
